src/client/reducers/districtDrawing.ts

```diff
@@ -259,10 +264,32 @@ const districtDrawingReducer: LoopReducer<ProjectState, Action> = (
         ...state,
         selectionTool: action.payload
       };
-    case getType(setGeoLevelIndex):
-      return updateCurrentState(state, {
-        geoLevelIndex: action.payload
-      });
+    case getType(setGeoLevelIndex): {
+      if (
+        !state.showAdvancedEditingModal &&
+        "resource" in state.staticData &&
+        "resource" in state.projectData
+      ) {
+        const [index, isReadOnly] = action.payload;
+        const { geoLevelHierarchy } = state.staticData.resource.staticMetadata;
+        const { advancedEditingEnabled } = state.projectData.resource.project;
+        const canSwitch = canSwitchGeoLevels(
+          state.undoHistory.present.state.geoLevelIndex,
+          index,
+          geoLevelHierarchy,
+          state.undoHistory.present.state.selectedGeounits
+        );
+
+        return !canSwitch
+          ? loop(state, Cmd.none)
+          : isReadOnly || advancedEditingEnabled
```

Comment:
```suggestion
          : isReadOnly || advancedEditingEnabled || index < geoLevelHierarchy.length - 1
```

Currently, if you attempt to toggle to a smaller geounit (e.g., `county` -> `blockGroup`), the modal is displayed even if you are not toggling to `block`. This modal should only be displayed toggling to `block`.